src/event/EventPluginHub.js

```diff
@@ -174,7 +180,14 @@ var EventPluginHub = {
    * @param {string} registrationName Name of listener (e.g. `onClick`).
    */
   deleteListener: function(id, registrationName) {
+    var PluginModule =
+      EventPluginRegistry.registrationNameModules[registrationName];
+    if (PluginModule && PluginModule.willDeleteListener) {
+      PluginModule.willDeleteListener(id, registrationName);
+    }
+
     var bankForRegistrationName = listenerBank[registrationName];
+    // TODO: This should never be null -- when is it?
```

Comment:
Hmm... when `onClick={null}` like you're trying to guard against in `ReactDOMComponent.js`?

What if instead of checking `lastProps[propKey]` in `ReactDOMComponent.js`, you only invoked `PluginModule.willDeleteListener` if `!!bankForRegistrationName`?